packages/components/table/src/table.vue

```diff
@@ -285,7 +285,7 @@ const ScrollBarPosition = ref<string>('left')
 const lastScrollLeft = ref(0)
 const sortOrder = ref([])
 const sortIcon = ref({})
-const data = ref([...props.items])
+const data = toRef(props, 'items')
```

Comment:
this is a readonly var so it should be better to have :
```typescript
const data = toRef(() => props.items);
```

Reply:
and I just realised why there was a copy of the data here. The sort !
So as you're mutating the value, you made a copy of it beforehand.
That could be costly, depending on the data size.
It could be worth having an option for that: allowsToMutate or not.

Reply:
there is a typo:
`PuikTableScrollBarPosistion`
should be
`PuikTableScrollBarPosition`
[table.ts#L16](https://github.com/PrestaShopCorp/puik/blob/main/packages/components/table/src/table.ts#L16)

Reply:
yes that's what I did but it generates errors for me, the function expects arguments however it seems to respect the signature as in the example in the doc:
```
// creates a readonly ref that calls the getter on .value access
toRef(() => props.foo)
```

I will use watch() as an alternative

Reply:
> and I just realised why there was a copy of the data here. The sort ! So as you're mutating the value, you made a copy of it beforehand. That could be costly, depending on the data size. It could be worth having an option for that: allowsToMutate or not.

yes, it can be expensive, otherwise there is a sortFromServer prop which prevents sorting on the client side. Only a SortOptions type payload is sent (which can be useful for building a server-side query instead)

Reply:
> yes that's what I did but it generates errors for me, the function expects arguments however it seems to respect the signature as in the example in the doc:
> 
> ```
> // creates a readonly ref that calls the getter on .value access
> toRef(() => props.foo)
> ```
> 
> I will use watch() as an alternative

ha! it must be because the project was still in 4.2.47...